### ISSUE TITLE
Remove dependency on external docker images

### ISSUE THREAD

**Issue by mirekphd** [elyra-ai/examples#38]
[This "call home"](https://github.com/elyra-ai/examples/blob/master/pipelines/dax_noaa_weather_data/analyze_NOAA_weather_data.pipeline#L17) (`docker pull` of the "Pandas" docker image from your Docker Hub account) will be unacceptable in any corporate security-restricted environment, where only certain security-approved docker containers / registries are permitted (and some users  work on completely air-gapped servers). You will also fail to execute any `docker pull` (if this is what you assumed to work), as this requires root-level priviledges.

Is there a truly local option - running in the same docker container where the Jupyter Lab is being executed? I mean it is completely false premise that it must be resource-poor local notebook. The client stays there, correct, but python kernel is nearly always run on a large compute node. So just let the user scripts (pipeline code payload) run on the same machine (but dedicated python kernel!) as the controlling notebook. This is how `papermill` works by the way (no additional docker pulls required).

**Comment by ptitzler** [comment]
> [... ] will be unacceptable in any corporate security-restricted environment [...]

You are absolutely right. To address this container images being used to run notebook or Python scripts can also be stored in private/local container registries, as mentioned here: https://elyra.readthedocs.io/en/latest/user_guide/runtime-image-conf.html#prerequisites. I just did notice [this in the documentation](https://elyra.readthedocs.io/en/latest/user_guide/runtime-image-conf.html#image-name-image-name), which we need to fix.

[...] if there is a truly local option - running in the same docker container (supplied and approved by the corporation) where Jupyter Lab is being executed?

Yes. The "local" runtime configuration is provided for that exact purpose. It is defined by default (unlike Kubeflow Pipelines configuration), and "local" refers to the machine where JupyterLab is running. We currently don't have this documented at https://elyra.readthedocs.io/en/latest/user_guide/runtime-conf.html, but probably should.

**Comment by ptitzler** [comment]
Transferring the issue to elyra-ai/elyra.
